src/components/remote_control/test/commands/set_interior_vehicle_data_request_test.cc

```diff
@@ -0,0 +1,411 @@
+
+using application_manager::MockService;
+using test::components::remote_control_test::MockApplication;
+
+using ::testing::_;
+using ::testing::Mock;
+using ::testing::NiceMock;
+using ::testing::StrictMock;
+using ::testing::Return;
+using ::testing::ReturnRef;
+using ::testing::SaveArg;
+using ::application_manager::Message;
+using ::application_manager::MessageType;
+using ::application_manager::ApplicationSharedPtr;
+using ::protocol_handler::MessagePriority;
+using remote_control::RemotePluginInterface;
+using remote_control::MessageHelper;
+using namespace remote_control;
+
+namespace {
+const int32_t kConnectionKey = 5;
+const int kModuleId = 153;
+const std::string kValidMobileRequestWithOnlySettableParams =
+    "{\"moduleData\":{\"moduleType\":\"CLIMATE\", "
+    "\"climateControlData\":{\"defrostZone\":"
+    "\"FRONT\", \"fanSpeed\":10}}}";
+const std::string kValidMobileRequestWithSettableAndReadOnlyParams =
+    "{\"moduleData\":{\"moduleType\":\"CLIMATE\", "
+    "\"climateControlData\":{\"defrostZone\":"
+    "\"FRONT\", \"fanSpeed\":10, \"currentTemperature\":{\"unit\":\"CELSIUS\", "
+    "\"value\":17.5}}}}";
+const std::string kValidMobileRequestWithoutSettableAndWithReadOnlyParams =
+    "{\"moduleData\":{\"moduleType\":\"CLIMATE\", \"climateControlData\":"
+    "{\"currentTemperature\":{\"unit\":\"CELSIUS\", \"value\":17.5}}}}";
+const std::string kInvalidMobileRequest = "{{\"moduleTip\":\"LUXOFT\"}}";
+const std::string kValidHmiResponse =
+    "{\"jsonrpc\":\"2.0\",\"id\":51,\"result\":{\"code\":0,\"method\":\"RC."
+    "SetInteriorVehicleData\",\"moduleData\":{\"moduleType\":\"CLIMATE\","
+    "\"climateControlData\":{\"fanSpeed\":0,\"desiredTemperature\":{\"unit\":"
+    "\"CELSIUS\","
+    "\"value\":25},\"acEnable\":true,\"circulateAirEnable\":true,"
+    "\"autoModeEnable\":true,\"defrostZone\":\"ALL\",\"dualModeEnable\":true,"
+    "\"acMaxEnable\":true,\"ventilationMode\":\"BOTH\"}}}}";
+const std::string kInvalidHmiResponse =
+    "{\"jsonrpc\":\"2.0\",\"id\":51,\"result\":{\"code\":21,\"method\":\"RC."
+    "SetInteriorVehicleData\",\"moduleData\":{\"moduleType\":\"CLIMATE\","
+    "\"ControlData\":{\"Speed\":0,\"outsideTemperature\":{\"unit\":"
+    "\"CELSIUS\",\"value\":\"high\"},\"desiredTemperature\":{\"unit\":"
+    "\"CELSIUS\","
+    "\"value\":25},\"acEnable\":true,\"circulateAirEnable\":true,"
+    "\"autoModeEnable\":true,\"defrostZone\":\"ALL\",\"dualModeEnable\":true,"
+    "\"acMaxEnable\":true,\"ventilationMode\":\"BOTH\"}},\"isSubscribed\":"
+    "false}}";
+}
+
+namespace test {
+namespace components {
+namespace remote_control_test {
+namespace set_interior_vehicle_data_request_test {
+
+class SetInteriorVehicleDataRequestTest : public ::testing::Test {
+ public:
+  typedef utils::SharedPtr<remote_control::commands::BaseCommandRequest>
+      SIVDRequestPtr;
+  typedef rc_event_engine::Event<application_manager::MessagePtr, std::string>
+      SIVD_HMI_Response;
```

Comment:
@Itileda response and event are different entities. Please do not mix this names. use event in typedef